src/engine/renderer/glsl_source/lighttile_fp.glsl

```diff
@@ -37,140 +37,121 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+	}
 }
 
-vec3 ProjToView(vec2 inp)
-{
-	vec3 p = u_zFar * vec3(inp, -1);
-	
-	return p;
+vec3 ProjToView( vec2 inp ) {
+	return u_zFar * vec3( inp, -1 );
 }
 
 void main() {
-  vec2 minmax = texture2D( u_DepthMap, 0.5 * vPosition + 0.5 ).xy;
-
-  float minx = vPosition.x - r_tileStep.x;
-  float maxx = vPosition.x + r_tileStep.x;
-  float miny = vPosition.y - r_tileStep.y;
-  float maxy = vPosition.y + r_tileStep.y;
-
-  vec3 bottomleft = ProjToView(vec2(minx, miny));
-  vec3 bottomright = ProjToView(vec2(maxx, miny));
-  vec3 topright = ProjToView(vec2(maxx, maxy));
-  vec3 topleft = ProjToView(vec2(minx, maxy));
-
-  vec4 plane1 = vec4(normalize(cross(bottomleft, bottomright)), 0);
-  vec4 plane2 = vec4(normalize(cross(bottomright, topright)), 0);
-  vec4 plane3 = vec4(normalize(cross(topright, topleft)), 0);
-  vec4 plane4 = vec4(normalize(cross(topleft, bottomleft)), 0);
-
-  vec4 plane5 = vec4( 0.0, 0.0,  1.0,  minmax.y );
-  vec4 plane6 = vec4( 0.0, 0.0, -1.0, -minmax.x );
-
-  idxs_t idxs = idx_initializer;
-
-  for( int i = u_lightLayer; i < u_numLights; i += numLayers ) {
-    light l = GetLight( i );
-    vec3 center = ( u_ModelMatrix * vec4( l.center, 1.0 ) ).xyz;
-    float radius = max( 2.0 * l.radius, 2.0 * 32.0 ); // Avoid artifacts with weak light sources
-
-    // todo: better checks for spotlights
-    lightOutsidePlane( plane1, center, radius );
-    lightOutsidePlane( plane2, center, radius );
-    lightOutsidePlane( plane3, center, radius );
-    lightOutsidePlane( plane4, center, radius );
-    lightOutsidePlane( plane5, center, radius );
-    lightOutsidePlane( plane6, center, radius );
-
-    if( radius > 0.0 ) {
-      pushIdxs( i, idxs );
-    }
-  }
-
-  exportIdxs( idxs );
+	vec2 minmax = texture2D( u_DepthMap, 0.5 * vPosition + 0.5 ).xy;
+
+	float minx = vPosition.x - r_tileStep.x;
+	float maxx = vPosition.x + r_tileStep.x;
+	float miny = vPosition.y - r_tileStep.y;
+	float maxy = vPosition.y + r_tileStep.y;
+
+	vec3 bottomleft = ProjToView( vec2( minx, miny ) );
+	vec3 bottomright = ProjToView( vec2( maxx, miny ) );
+	vec3 topright = ProjToView( vec2( maxx, maxy ) );
+	vec3 topleft = ProjToView( vec2( minx, maxy ) );
+
+	vec4 plane1 = vec4( normalize( cross( bottomleft, bottomright ) ), 0 );
+	vec4 plane2 = vec4( normalize( cross( bottomright, topright ) ), 0 );
+	vec4 plane3 = vec4( normalize( cross( topright, topleft ) ), 0 );
+	vec4 plane4 = vec4( normalize( cross( topleft, bottomleft ) ), 0 );
+
+	vec4 plane5 = vec4( 0.0, 0.0,  1.0,  minmax.y );
+	vec4 plane6 = vec4( 0.0, 0.0, -1.0, -minmax.x );
+
+	idxs_t idxs = uvec4( 0, 0, 0, 0 );
+
+	// NOT the amount of lights that can actually be put into each layer of the lighttile texture
+	uint globalLightsPerLayer = min( ( u_numLights + numLayers - 1 ) / numLayers, lightsPerLayer );
+	uint lightOffset = u_lightLayer * globalLightsPerLayer;
+	uint layerLightCount = lightOffset < u_numLights ? min( u_numLights - lightOffset, globalLightsPerLayer ) : 0;
```

Comment:
This looks wrong: it makes it so that no more than 16 lights are ever considered. There are up to 256 input lights and they should all be considered as long as less than 16 lights have been output after culling.

Reply:
Fixed.